pkg/controllers/managementuser/resourcequota/resource_quota_sync.go

```diff
@@ -244,39 +254,15 @@ func (c *SyncController) getExistingLimitRange(ns *corev1.Namespace) (*corev1.Li
 	return limitRanger[0], nil
 }
 
-func (c *SyncController) getNamespaceResourceQuota(ns *corev1.Namespace) (*corev1.ResourceQuotaSpec, error) {
+func (c *SyncController) getNamespaceResourceQuota(ns *corev1.Namespace) (v32.NamespaceResourceQuota, error) {
```

Comment:
Why did the return value change from a pointer to non-pointer?

Reply:
>  There were a few places where the code returns nil, nil for the value and error, which the caller may not properly handle.

I disagree with this reasoning. If a method returns a pointer, it is absolutely the caller's responsibility to check for nilness before using it.

> But then I also thought it's fine to keep the change - I return the value itself, rather then the pointer, from the function which creates the value.

A nil value is useful for showing whether the value has been set and whether it makes sense to use the value. In this case, the point of this nil check

https://github.com/rancher/rancher/pull/36652/files#diff-6247a4224a746023f729c575824523aa859221464fb312fc067de65baf55270dR381

is diluted because the object received from 

https://github.com/rancher/rancher/pull/36652/files#diff-6247a4224a746023f729c575824523aa859221464fb312fc067de65baf55270dR121

is never nil, so the default quota will never be used.